docs/jsx.md

```diff
@@ -0,0 +1,101 @@
+# JSX
+
+Framework supports [React](https://react.dev/) and [JSX](https://react.dev/learn/writing-markup-with-jsx). This provides a convenient way of writing dynamic HTML in JavaScript, using React to manage state and rendering. To use JSX, declare a JSX fenced code block (<code>```jsx</code>), and then call the built-in display function to display some content.
+
+````md
+```jsx
+display(<i>Hello, <b>JSX</b>!</i>);
+```
+````
+
+This produces:
+
+```jsx
+display(<i>Hello, <b>JSX</b>!</i>);
+```
+
+JSX is especially convenient for authoring reusable components. These components are typically imported from JSX modules (`.jsx`), but you can also declare them within JSX fenced code blocks.
+
+```jsx echo
+function Greeting({subject = "you"} = {}) {
+  return <div>Hello, <b>{subject}</b>!</div>
+}
+```
+
+Naturally, you can combine JSX with Framework’s built-in reactivity. This is typically done by passing in reactive values as props. Try changing the `name` below.
+
+```jsx echo
+display(<Greeting subject={name} />);
+```
+
+```js echo
+const name = view(Inputs.text({label: "Name", value: "Anonymous"}));
+```
+
+Below we import a JSX component and render it.
+
+```jsx echo
+import {Counter} from "./components/Counter.js";
+
+display(<Counter title="Hello, JSX" />);
+```
+
+With a JSX fenced code block, the [display function](./javascript#explicit-display) behaves a bit differently:
+
+- It replaces the previously-displayed content, if any
+- It never uses the inspector
+
+In addition, JSX fenced code blocks should always display explicitly; JSX fenced code blocks do not support implicit display of expressions.
+
+You don’t need to import `React` when using JSX; `React` is implicitly imported. `React` is also available by default in Markdown. If you need to, you can import it explicitly:
+
+```js run=false
+import * as React from "npm:react";
+```
+
+You can also import specific symbols such as hooks:
+
+```js run=false
+import {useState} from "npm:react";
+```
+
+Always use the `.js` extension to import JSX modules.
```

Comment:
```suggestion
Always use the `.js` file extension to import JSX modules.
```

(suggestion to slightly reinforce that extension is the file extension not the fenced block language)

Reply:
I’m referring to the file extension here. You import `Card.js` even though the source is `Card.jsx`. That’s because the JSX gets transpiled to JS and served as `Card.js`; at runtime it doesn’t matter that it was originally authored in JSX; at runtime it’s JS. (And if you’re a masochist you could hand-author components in JavaScript if you wanted.) The same convention is used to import TypeScript; you import `.js` even though the file is `.ts`.